File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        for (int i = 0; i < nums.length; i++){
            mp.put(nums[i], i);
        }
        if (nums.length == mp.size()) return false;
        else return true;
    }
}

File: Data Structures & Algorithms/target-sum/submission-1.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        n = len(nums)

        def btrack(index, remainder):
            if index == n and remainder == target:
                return 1
            if index == n and remainder != target:
                return 0
            
            return btrack(index+1, remainder+nums[index])+btrack(index+1, remainder-nums[index])
        
        return btrack(0, 0)